addons/website_sale/models/product.py

```diff
@@ -1,6 +1,8 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 from odoo import api, fields, models, tools, _
+from odoo.http import request
```

Comment:
not really happy about this. this is mixing controller logic with model/business logic